From 8342ed803d6fb43675168b6ebb88a3b8a688dda2 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Tue, 23 Aug 2022 12:25:57 +0300
Subject: [PATCH] [SW-99461] gaudi3: add backup image descriptor data

In preparation for backup of boot info usage in rom, descriptor
data that is required in case of unavailability of boot info
is added in this patch.

Change-Id: I2e755229b1795c1095c27aa4efa1cf49c65bee12
---

diff --git a/include/arcboot.h b/include/arcboot.h
index 6141a5d..87d5323 100644
--- a/include/arcboot.h
+++ b/include/arcboot.h
@@ -77,6 +77,7 @@
 #define CRC32_SIZE		4
 #define SYS_REF_CLOCK				50000000ULL
 
+#define MAX_PERSISTENT_IMG_DESC		8
 struct arcboot_global_s {
 	uint32_t config_fuse;
 	uint32_t config2_fuse;
@@ -91,6 +92,8 @@
 	struct hl_fw_boot_ptrs_ctx boot_ptr_ctx[HL_FW_BOOT_PTR_MAX_IMG];
 	struct hl_fw_sec_info sec_data;
 	uint8_t tpm_validation_pass;
+	struct hl_fw_img_desc spi_img_desc[MAX_PERSISTENT_IMG_DESC];
+	struct hl_fw_boot_ptrs spi_manifest_hdr;
 };
 
 enum arcboot_rom_bringup_proggress {
